### ISSUE TITLE
Updating e2e test binary and k8s components without rebuilding and restarting the entire e2e cluster.

### ISSUE THREAD

**Issue by verult** [kubernetes/test-infra#6699]
/kind feature

While modifying and iterating over e2e tests, I'd like to be able to quickly build and run it on my existing e2e clusters. Ideally I can do this by only rebuilding the test binary and dropping it in the cluster. Today, I have to do `make quick-release` in parallel with `kubetest --down`, and then `kubetest --up --test --test_args...`. When a test fails due to a bug in the code, I'd like to do the same thing but for that bug: rebuild the affected k8s binary, drop it in the cluster, and re-run the e2e test. I couldn't find any commands/scripts or manual instructions on how to do this.

It would speed up the development cycle a lot if we have a better, more visible process for this. Thanks!

**Comment by verult** [comment]
Turns out there are instructions at the top of [this section](https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md#building-kubernetes-and-running-the-tests) of the e2e test guide with building only the test binaries using `make WHAT=test/e2e/e2e.test`.

The other half of the request still applies: it would be great if there is some guidance on quickly updating a binary in the cluster.

**Comment by verult** [comment]
/remove-lifecycle rotten

Pinging again. So far I've been resorting to my own bash scripts, but it would be nice to have a standard way to do something like this. Happy to chat about it more.

**Comment by verult** [comment]
Thinking about it more - this is possibly out of scope for test-infra. Will close this issue.
